docs/resources/server/enchantments.md

```diff
@@ -0,0 +1,319 @@
+```json
+"minecraft:attributes": [
+    {
+        "amount": {
+            "type": "minecraft:linear",
+            "base": 1,
+            "per_level_above_first": 1
+        },
+        "attribute": "namespace:class.attribute_name",
+        "id": "examplemod:enchantment.example",
+        "operation": "add_multiplied_base"
+    }
+],
+```
+
+The object within the `amount` block is a [Level Based Value], which can be used to have a Value Effect Component that changes the intensity of its effect by level. The `operation` is one of `add_value`, `add_multiplied_base` or `add_multiplied_total`. See [Attribute Operations] for details.
+
+### Value Effect Components
+[Value Effect Components] are used for enchantments that alter a numerical value somewhere in the game, and are implemented by the class `EnchantmentValueEffect`. 
+
+Value Effect Components can be set to use any of these operations on their given values:
+- `minecraft:set`: Overwrites the given level-based value.
+- `minecraft:add`: Adds the specified level-based value to the old one.
+- `minecraft:multiply`: Multiplies the specified level-based factor by the old one.
+- `minecraft:remove_binomial`: Polls a given (level-based) chance using a binomial distibution. If it works, subtracts 1 from the value. Note that many values are effectively flags, being fully on at 1 and fully off at 0.
+- `minecraft:all_of`: Accepts a list of other value effects and applies them in the stated sequence.
+
+An example from Vanilla is `minecraft:damage`, which allows for an enchantment to modify the attack damage done to an entity when an enchanted item is used as a weapon. The Sharpness enchantment uses this component as follows to achieve its effect:
+```json
+"effects": {
+    "minecraft:damage": [
+      {
+        "effect": {
+          "type": "minecraft:add",
+          "value": {
+            "type": "minecraft:linear",
+            "base": 1.0,
+            "per_level_above_first": 0.5
+          }
+        }
+      }
+    ]
+  }
+```
+
+#### Using Enchantment Value Effect Components
+One way to adjust values based on custom Value Effect components is to invoke one of the overloads of `EnchantmentHelper#runIterationOnItem`. This function accepts an `EnchantmentHelper.EnchantmentVisitor`, which is a functional interface that accepts an enchantment and its level, and is invoked on all of the enchantments that the given itemstack has. While any consumer of those two values will work, the `Enchantment` class provides a handy function that fits this interface (and is used often by vanilla) -- `Enchantment#applyEffects`. This function takes a `List<ConditionalEffect<EnchantmentValueEffect>>`, a `LootContext`, and a function that consumes a value effect. The function verifies that all of the `ConditionalEffect`s are satisfied given the provided `LootContext`, and if so, invokes the provided function.
```

Comment:
I feel like this a bit too brief to understand what is occurring. Adjusting the value is an iterative process where the the mutable value is continually updated by each successive enchantment effect applied. This means that multipliers have the potential to grow exponentially depending on the circumstance.

There needs to be more detail to understand what's going on, and how it compounds or relates to the final returned value.